### PR TITLE
AN-87 Add table support

### PR DESCRIPTION
**Pull request by kevinfodness**
* Adds table tags to the whitelist of allowed tags.
* Adds a new `Table` component to handle HTML tables.
* Adds a test to confirm the functionality of the table component.

### CODE REVIEW

File: includes/apple-exporter/components/class-table.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	/**
+	 * Look for node matches for this component.
+	 *
+	 * @param \DOMNode $node The node to inspect for matches.
+	 *
+	 * @access public
+	 * @return \DOMNode|null The DOMNode if there is a match, null if not.
+	 */
+	public static function node_matches( $node ) {
+
```
Comment (jakewrfoster) [inline includes/apple-exporter/components/class-table.php]: Not a big deal, however, some phpcs config complain about this empty line

File: includes/apple-exporter/components/class-table.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	/**
+	 * Build the component.
+	 *
+	 * @param string $html The html to convert into a component.
+	 *
+	 * @access protected
+	 */
+	protected function build( $html ) {
+
```
Comment (jakewrfoster) [inline includes/apple-exporter/components/class-table.php]: ditto

File: includes/apple-exporter/components/class-table.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+namespace Apple_Exporter\Components;
+
+/**
+ * A component class representing a table.
+ *
+ * @since 1.4.0
+ */
+class Table extends Component {
+
+	/**
```
Comment (jakewrfoster) [inline includes/apple-exporter/components/class-table.php]: are the lines within the class definition padded with spaces? It's hard to tell in GH sometimes what is a space and what is a tab.

File: includes/apple-exporter/components/class-table.php (hunk 1 above)
Comment (kevinfodness) [reply includes/apple-exporter/components/class-table.php]: Interesting. It looks weird to me if it's not spaced out like this, but that's certainly a personal preference. We're using a break like this in other places in the code. I've got a separate ticket to bring this entire repo up to modern wpcs coding standards, so if the config complains at that time, I'll fix it globally. Thanks!

File: includes/apple-exporter/components/class-table.php (hunk 3 above)
Comment (kevinfodness) [reply includes/apple-exporter/components/class-table.php]: I just checked, and it's tabs all the way down.

🌍 
🐢 
🐢 
🐢

**Pull request merged**